File: android/src/main/java/io/agora/agorartcengine/RtcUtil.java
package io.agora.agorartcengine;

import android.os.Environment;

import com.huawei.rtc.models.LogInfo;
import com.huawei.rtc.models.UserInfo;

import java.io.File;
import java.util.Map;

/**
 * 工具类
 *
 * @author zhengxiaobin
 * @date 2020/6/5
 */
class RtcUtil {

    static UserInfo userInfoFromJson(Map<String, Object> json) {
        UserInfo userInfo = new UserInfo();
        userInfo.setUserId((String) json.get("userId"));
        userInfo.setCtime((int) json.get("ctime"));
        userInfo.setOptionalInfo((String) json.get("optionalInfo"));
        int role = (int) json.get("role");
        userInfo.setRole(UserInfo.RoleType.values()[role]);
        userInfo.setSignature((String) json.get("signature"));
        userInfo.setUserName((String) json.get("userName"));
        return userInfo;
    }

    static LogInfo logInfoFromJson(Map<String, Object> json) {
        LogInfo userInfo = new LogInfo();
        userInfo.setPath((String) json.get("path"));
        int level = (int) json.get("level");
        userInfo.setLevel(LogInfo.LogLevel.values()[level]);
        return userInfo;
    }

    public static String getLogPath(){
        File path = Environment.getExternalStorageDirectory();
        return path.getAbsolutePath()+"/rtcLog";
    }

}

File: android/src/main/java/io/agora/agorartcengine/RtcUtil.kt
package io.agora.agorartcengine

import android.os.Environment
import com.huawei.rtc.models.LogInfo
import com.huawei.rtc.models.LogInfo.LogLevel
import com.huawei.rtc.models.UserInfo

/**
 * 工具类
 *
 * @author zhengxiaobin
 * @date 2020/6/5
 */
internal object RtcUtil {
    fun userInfoFromJson(json: Map<String, Any>): UserInfo {
        val userInfo = UserInfo()
        userInfo.userId = json["userId"] as String?
        userInfo.ctime = json["ctime"] as Int
        userInfo.optionalInfo = json["optionalInfo"] as String?
        val role = json["role"] as Int
        userInfo.role = UserInfo.RoleType.values()[role]
        userInfo.signature = json["signature"] as String?
        userInfo.userName = json["userName"] as String?
        return userInfo
    }

    fun logInfoFromJson(json: Map<String?, Any?>): LogInfo {
        val userInfo = LogInfo()
        userInfo.path = json["path"] as String?
        val level = json["level"] as Int
        userInfo.level = LogLevel.values()[level]
        return userInfo
    }

    val logPath: String
        get() {
            val path = Environment.getExternalStorageDirectory()
            return path.absolutePath + "/rtcLog"
        }
}